pkg/sign/certificate.go

```diff
@@ -220,3 +241,27 @@ func (f *Fulcio) GetCertificate(ctx context.Context, keypair Keypair, opts *Cert
 
 	return certBlock.Bytes, nil
 }
+
+// subjectFromClaims returns the claim to use for proof of key possession
+// signature.
+//
+// FIXME: this is taken from sigstore/sigstore. We don't currently use the
+// exported SubjectFromToken function because it requires an *oidc.IDToken
+// object, which is generated after the token is verified by the client.
+// In this library, the token is not verified as Fulcio also verifies the
+// token's signature anyway. Also, sigstore/sigstore's implementation of
+// SubjectFromToken operates on the token's bytes (tok.Claims() wraps around
+// json.Unmarshal), so possibly this change belongs there.
+func subjectFromClaims(c jsonWebToken) (string, error) {
```

Comment:
I'd be OK with a second method in sigstore/sigstore, something like `SubjectFromUnverifiedToken`.

@woodruffw I feel like this came up before in sigstore-python?

Reply:
Added in https://github.com/sigstore/sigstore/pull/1826